app/models/application_record.rb

```diff
@@ -7,6 +7,13 @@ class ApplicationRecord < ActiveRecord::Base
   after_update { |record| log_analytics record, "updated" }
   after_destroy { |record| log_analytics record, "destroyed" }
 
+  # Condition for rollout of mass addition of validation rules.
+  def mass_validation_enabled?
+    Rails.env.development? ||
+      Rails.env.test? ||
+      AppConfig.find_by(key: AppConfig::ENABLE_MASS_VALIDATION)
```

Comment:
I'd say we need to optimize this method, it can become slow during run_time. Remember this code will be invoked for every field, and it involves a query to the database.
Maybe we should just have a constant loaded when you start rails, and changes in this field would only require a restart of the container.

Reply:
@davidrissato  Good point. Will do. 

@tfrcarvalho you had some comment about removing the `Rails.env` dependency and relying only on `AppConfig` is that right?

Reply:
Yes, and actually for test - in theory at least - you should run all the relevant tests with the flag both enabled and disabled, to test current and future behavior. 
You can do loops where you run the same tests and enable/disable the flag.

Reply:
Ok I made the change and I tested locally where `ENABLE_MASS_VALIDATION` is both true and false.